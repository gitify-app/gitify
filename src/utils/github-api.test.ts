import type { StateType, Subject, SubjectType } from '../typesGithub';
import {
  getNotificationTypeIcon,
  getNotificationTypeIconColor,
} from './github-api';

describe('utils/github-api.ts', () => {
  describe('getNotificationTypeIcon - should get the notification type icon', () => {
    expect(
      getNotificationTypeIcon(
        createSubjectMock({ type: 'CheckSuite', state: null }),
      ).displayName,
    ).toBe('RocketIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'CheckSuite',
          state: 'cancelled',
        }),
      ).displayName,
    ).toBe('StopIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'CheckSuite',
          state: 'failure',
        }),
      ).displayName,
    ).toBe('XIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'CheckSuite',
          state: 'skipped',
        }),
      ).displayName,
    ).toBe('SkipIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'CheckSuite',
          state: 'success',
        }),
      ).displayName,
    ).toBe('CheckIcon');
    expect(
      getNotificationTypeIcon(createSubjectMock({ type: 'Commit' }))
        .displayName,
    ).toBe('GitCommitIcon');
    expect(
      getNotificationTypeIcon(createSubjectMock({ type: 'Discussion' }))
        .displayName,
    ).toBe('CommentDiscussionIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({ type: 'Discussion', state: 'DUPLICATE' }),
      ).displayName,
    ).toBe('DiscussionDuplicateIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({ type: 'Discussion', state: 'OUTDATED' }),
      ).displayName,
    ).toBe('DiscussionOutdatedIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({ type: 'Discussion', state: 'RESOLVED' }),
      ).displayName,
    ).toBe('DiscussionClosedIcon');
    expect(
      getNotificationTypeIcon(createSubjectMock({ type: 'Issue' })).displayName,
    ).toBe('IssueOpenedIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({ type: 'Issue', state: 'draft' }),
      ).displayName,
    ).toBe('IssueDraftIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'Issue',
          state: 'closed',
        }),
      ).displayName,
    ).toBe('IssueClosedIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'Issue',
          state: 'completed',
        }),
      ).displayName,
    ).toBe('IssueClosedIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'Issue',
          state: 'not_planned',
        }),
      ).displayName,
    ).toBe('SkipIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'Issue',
          state: 'reopened',
        }),
      ).displayName,
    ).toBe('IssueReopenedIcon');
    expect(
      getNotificationTypeIcon(createSubjectMock({ type: 'PullRequest' }))
        .displayName,
    ).toBe('GitPullRequestIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'PullRequest',
          state: 'draft',
        }),
      ).displayName,
    ).toBe('GitPullRequestDraftIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'PullRequest',
          state: 'closed',
        }),
      ).displayName,
    ).toBe('GitPullRequestClosedIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'PullRequest',
          state: 'merged',
        }),
      ).displayName,
    ).toBe('GitMergeIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'Release',
        }),
      ).displayName,
    ).toBe('TagIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'RepositoryInvitation',
        }),
      ).displayName,
    ).toBe('MailIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'RepositoryVulnerabilityAlert',
        }),
      ).displayName,
    ).toBe('AlertIcon');
    expect(
      getNotificationTypeIcon(
        createSubjectMock({
          type: 'WorkflowRun',
        }),
      ).displayName,
    ).toBe('RocketIcon');
    expect(getNotificationTypeIcon(createSubjectMock({})).displayName).toBe(
      'QuestionIcon',
    );
  });

  describe('getNotificationTypeIconColor', () => {
    it('should format the notification color for check suite', () => {
      expect(
        getNotificationTypeIconColor(
          createSubjectMock({
            type: 'CheckSuite',
            state: 'cancelled',
          }),
        ),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(
          createSubjectMock({
            type: 'CheckSuite',
            state: 'failure',
          }),
        ),
      ).toMatchSnapshot();

      expect(
        getNotificationTypeIconColor(
          createSubjectMock({
            type: 'CheckSuite',
            state: 'skipped',
          }),
        ),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(
          createSubjectMock({
            type: 'CheckSuite',
            state: 'success',
          }),
        ),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(
          createSubjectMock({
            type: 'CheckSuite',
            state: null,
          }),
        ),
      ).toMatchSnapshot();
    });

    it('should format the notification color for state', () => {
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'ANSWERED' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'closed' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'completed' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'draft' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'merged' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(
          createSubjectMock({ state: 'not_planned' }),
        ),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'open' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'reopened' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(createSubjectMock({ state: 'RESOLVED' })),
      ).toMatchSnapshot();
      expect(
        getNotificationTypeIconColor(
          createSubjectMock({
            state: 'something_else_unknown' as StateType,
          }),
        ),
      ).toMatchSnapshot();
    });
  });
});

function createSubjectMock(mocks: {
  title?: string;
  type?: SubjectType;
  state?: StateType;
}): Subject {
  return {
    title: mocks.title ?? 'Mock Subject',
    type: mocks.type ?? ('Unknown' as SubjectType),
    state: mocks.state ?? ('Unknown' as StateType),
    url: null,
    latest_comment_url: null,
  };
}
