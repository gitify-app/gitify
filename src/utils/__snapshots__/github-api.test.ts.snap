// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for check suite 1`] = `"text-gray-500 dark:text-gray-300"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for check suite 2`] = `"text-red-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for check suite 3`] = `"text-gray-500 dark:text-gray-300"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for check suite 4`] = `"text-green-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for check suite 5`] = `"text-gray-500 dark:text-gray-300"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 1`] = `"text-green-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 2`] = `"text-red-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 3`] = `"text-purple-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 4`] = `"text-gray-500 dark:text-gray-300"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 5`] = `"text-purple-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 6`] = `"text-gray-500 dark:text-gray-300"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 7`] = `"text-green-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 8`] = `"text-green-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 9`] = `"text-purple-500"`;

exports[`utils/github-api.ts getNotificationTypeIconColor should format the notification color for state 10`] = `"text-gray-500 dark:text-gray-300"`;
